src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrderCheck.java

```diff
@@ -0,0 +1,365 @@
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.ListIterator;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * <p>
+ * Checks that the parameter order in javadoc is matching with the parameter order in method.
+ * It includes both method parameter and type parameters of generic method.
+ * This check expects the user to provide same number of parameter tag in javadoc as
+ * parameters in method.
+ * </p>
+ * <p>
+ * Examples:
+ * </p>
+ * <ul>
+ * <li>
+ * Example of method parameters:
+ * <pre>
+ * class methodParams {
+ *   &#47;**
+ *    * Description
+ *    *
+ *    * &#64;param input description
+ *    * &#64;param output desc
+ *    *&#47;
+ *   public void method(String input, int output) {
+ *   }
+ * }
+ * </pre>
+ * </li>
+ * <li>
+ * Example of type parameters of generic method:
+ * <pre>
+ * class genericMethodParams {
+ *   &#47;**
+ *    * Description.
+ *    *
+ *    * &#64;version Some number
+ *    * &#64;param &lt;T&gt; description
+ *    * &#64;param p1 desc
+ *    * &#64;return p1 desc
+ *    * &#64;see more link.
+ *    *&#47;
+ *   public &lt;T&gt; String genericMethod (T p1) {
+ *     return p1;
+ *   }
+ * }
+ * </pre>
+ * </li>
+ * </ul>
+ * <p>
+ * Notes:
+ * </p>
+ * <ul>
+ * <li>
+ * This check will only report if the parameter in method did not match the
+ * parameter in javadoc.
+ * </li>
+ * <li>
+ * This check will expect from user to provide same number of parameter in method
+ * and in javadoc.
+ * </li>
+ * <li>
+ * This check will not report any violations for any other tag in between param
+ * tags in javadoc.
+ * </li>
+ * <li>
+ * This check will not report any violations for missing parameter in javadoc.
+ * </li>
+ * <li>
+ * This check will not report any violations for extra parameter in javadoc.
+ * </li>
+ * </ul>
+  * <ul>
+ * <li>
+ * Property {@code violateExecutionOnNonTightHtml} - Control when to print violations if the
+ * Javadoc being examined by this check violates the tight html rules defined at
+ * <a href="https://checkstyle.org/writingjavadocchecks.html#Tight-HTML_rules">Tight-HTML Rules</a>.
+ * Type is {@code boolean}.
+ * Default value is {@code false}.
```

Comment:
Please add unit tests exercising this property, if we are going to add it. I am not sure that `@param` tags will really use HTML, but this help someone avoid parsing issues.

Reply:
i am not sure about what/how to do this, please help.

Reply:
On second thought, I see that we have test coverage for this property in `AbstractJavadocCheckTest.java`, so we can skip this here. Resolved.